### DISCUSSION TITLE
Creating Public Variable with Proxy Software Catalog Item

### DISCUSSION THREAD

**Discussion by swageblock** [mia-platform/community#584] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

13.4

### Description

I am attempting to create a software catalog item that will add an endpoint on a specific listener , a proxy, and public variable for outbound traffic from the Envoy api-gateway.  These share an infrastructure-layer api gateway (Azure API Management) that we are targeting sub-resources on.  If Project A and B need access to Project C functionality, this is all routed through a shared api gateway instance with sub-resources indicating A, B, and C:

example: `http://apigw/A, http://apigw/B, http://apigw/C`

### Actual Outcome

What I see with the software catalog for a proxy-type when I specify unsecreted variables is that the variable is not created:
```{
    "services": {
        "Capability-Proxy": {
            "type": "external",
            "name": "{capability-initial}-proxy",
            "url": "http://{{VU_AZURE_APIM}}/{capability-initial}"
        }
    },
    "unsecretedVariables": {
        "VU_AZURE_APIM": {
            "productionEnv": "internalK8sAddress",
            "noProductionEnv": "internalK8sAddress"
        }
    }
}
```
What I see when I attempt to use an application-type it seems to be that the service, although defined as an `external`, the console is attempting to create a repo in Github (I assume leveraging the provider configuration that we have configured).

### Expected Outcome

I would like to create a proxy (using a public variable), an endpoint for this proxy, and the public variable be created if it doesn't exist.

**Comment by galatifederico** [comment]
Hi @swageblock,
At the moment, it is not possible to create a proxy with the application. Our support team will contact you to better understand your needs and the priority related to this case.
Thank you.

**Reply by TheRusCompass** [reply, marked as answer]
Hi @swageblock ,
at the moment we do not have any plan to add this feature in our product evolution. If it is still a requirement on your side please let us know.
Thank you!
